apps/studio/components/interfaces/Auth/ThirdPartyAuthForm/CreateAuth0Dialog.tsx

```diff
@@ -0,0 +1,182 @@
+import { zodResolver } from '@hookform/resolvers/zod'
+import { Trash } from 'lucide-react'
+import { useEffect } from 'react'
+import { SubmitHandler, useForm } from 'react-hook-form'
+import toast from 'react-hot-toast'
+import * as z from 'zod'
+
+import { useParams } from 'common'
+import { useCreateThirdPartyAuthIntegrationMutation } from 'data/third-party-auth/integration-create-mutation'
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogSection,
+  DialogTitle,
+  FormControl_Shadcn_,
+  FormField_Shadcn_,
+  Form_Shadcn_,
+  Input_Shadcn_,
+  Separator,
+} from 'ui'
+import { FormItemLayout } from 'ui-patterns/form/FormItemLayout/FormItemLayout'
+
+interface CreateAuth0IntegrationProps {
+  visible: boolean
+  onClose: () => void
+  // TODO: Remove this if this Dialog is only used for creating.
+  onDelete: () => void
+}
+
+const FORM_ID = 'create-auth0-auth-integration-form'
+
+const FormSchema = z.object({
+  enabled: z.boolean(),
+  auth0DomainName: z
+    .string()
+    .trim()
+    .min(1)
+    .regex(/^[A-Za-z0-9-.]+$/, 'The project ID contains invalid characters.'), // Only allow alphanumeric characters and hyphens.
```

Comment:
let's raise the exact error up for the user?

```suggestion
    .regex(/^[A-Za-z0-9-.]+$/, 'Project IDs should only have alphanumeric characters and hyphens.'),
```

Reply:
Fixed.